2-classes-db/task.md

```diff
@@ -21,31 +21,43 @@ To submit this homework write the correct commands after each question.
 
 ### 1. Show the date, transaction_no, description and amount for those transactions whose amount is between £30,000 and £31,000.
 ```sql
-
+select date, transaction_no, description, amount from spends
+where amount > 30000 and amount < 31000
+ 
 ```
 ### 2. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee'.
 ```sql
-
+select date, transaction_no, supplier_inv_no, description, amount from spends
+where  description like '%fee%'
 ```
 ### 3. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'Fee'.
 ```sql
-
+select date, transaction_no, supplier_inv_no, description, amount from spends
+where  description like '%Fee%'
 ```
 ### 4. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee' (case insensitive). You will need to search 'https://www.postgresql.org/docs/' to solve this.
 ```sql
-
+select date, transaction_no, supplier_inv_no, description, amount from spends
+where  lower (description) like '%fee%'
 ```
 ### 5. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose amount is £25,000, £30,000, £35,000 or £40,000.
 ```sql
-
+select date, transaction_no, supplier_inv_no, description, amount from spends
+where amount in (25000 , 30000 , 35000 , 40000) 
 ```
 ### 6. Show the date, the supplier_id, the description and the amount for transactions with the expense area of 'Better Hospital Food'. You could do a query to get the expense_area_id first then do a query to find the dates, supplier_ids and amounts. But it would be better to do this all in one query by linking the tables together using INNER JOINs.
 ```sql
+select spends.date, spends.supplier_id, spends.description, spends.amount from spends, expense_areas
+where expense_areas.expense_area = 'Better Hospital Food' and expense_areas.id = spends.expense_area_id
```

Comment:
Whilst this is not wrong, and should work, the instruction suggests using INNER JOIN.  An example might be:

```
SELECT spends.date, spends.supplier_id, spends.description, spends.amount 
FROM spends
INNER JOIN expense_areas ON spends.expense_area_id = expense_areas.id
WHERE expense_areas.expense_area = 'Better Hospital Food'
```

Reply:
Thank you @Alexander-Rennie.
I have not used much INNER JOIN :) Will start using it
Result is same and performance almost the same too.

Your script could be simplified, because we extracting data only from single table "spends" but probably it will became not so readable:

SELECT date, supplier_id, description, amount 
FROM spends
INNER JOIN expense_areas ON expense_area_id = expense_areas.id
WHERE expense_areas.expense_area = 'Better Hospital Food'